drivers/dac/dac_dacx0501.c

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2023 Google LLC.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief Driver for Texas Instruments DACx0501 series
+ *
+ * Device driver for the  Texas Instruments DACx0501 series of devices: DAC60501, DAC70501 and
+ * DAC80501: Digital to Analog Converters with a single channel output and with 12, 14 and 16
+ * bits resolution respectively. Data sheet can be found here:
+ * https://www.ti.com/lit/ds/symlink/dac80501.pdf
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/dac.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/byteorder.h>
+
+LOG_MODULE_REGISTER(dac_dacx0501, CONFIG_DAC_LOG_LEVEL);
+
+#define DACX0501_REG_DEVICE_ID 0x01U
+#define DACX0501_REG_SYNC      0x02U
+#define DACX0501_REG_CONFIG    0x03U
+#define DACX0501_REG_GAIN      0x04U
+#define DACX0501_REG_TRIGGER   0x05U
+#define DACX0501_REG_STATUS    0x07U
+#define DACX0501_REG_DAC       0x08U
+
+#define DACX0501_MASK_DEVICE_ID_RES	 GENMASK(14, 12)
+#define DACX0501_MASK_CONFIG_REF_PWDWN	 BIT(8)
+#define DACX0501_MASK_CONFIG_DAC_PWDWN	 BIT(0)
+#define DACX0501_MASK_GAIN_BUFF_GAIN	 BIT(0)
+#define DACX0501_MASK_GAIN_REFDIV_EN	 BIT(8)
+#define DACX0501_MASK_TRIGGER_SOFT_RESET (BIT(1) | BIT(3))
+#define DACX0501_MASK_STATUS_REF_ALM	 BIT(0)
+
+/* Specifies the source of the reference voltage. */
+enum RefVoltageSource {
+	REF_INTERNAL, /* Internal 2.5V voltage reference. */
+	REF_EXTERNAL, /* External pin voltage reference. */
+};
+
+/* Specifies the reference voltage multiplier. */
+enum VoltageMult {
+	VM_MUL2, /* Multiplies by 2. */
+	VM_NOP,	 /* Multiplies by 1. */
```

Comment:
I'd suggest to use the following for consistency:
```suggestion
	VM_MUL1,	 /* Multiplies by 1. */
```

Reply:
Done